File: little-bee/controller/router/vai.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_VAI = "/config/vai"
const URL_ROUTER_VAI_BY_ID = "/config/vai/:vai-id"
const URL_ROUTER_VAI_QUERY = "/config/vai/query"
const URL_ROUTER_VAI_ALL = "/config/vai/all"

func VaiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_VAI, config.CreateVai)
	r.GET(URL_ROUTER_VAI_BY_ID, config.RetrieveVai)
	r.PUT(URL_ROUTER_VAI, config.UpdateVai)
	r.DELETE(URL_ROUTER_VAI_BY_ID, config.DeleteVai)
	r.PUT(URL_ROUTER_VAI_BY_ID, config.DeleteVaiWithFlag)
	return r
}

File: frontend/console/src/api/vai.js
import Axios from "@/http";

export function createVai(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/vai', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveVai(vaiId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/vai/${ vaiId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateVai(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/vai', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteVai(vaiId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/vai/${ vaiId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteVaiWithFlag(vaiId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/vai/${ vaiId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllVais() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/vai/all').then(res => {
            resolve(res)
        })
    })
}

export function queryVai (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/vai/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}